apps/src/codebridge/InfoPanel/ValidationStatusIcon.tsx

```diff
@@ -0,0 +1,40 @@
+import classNames from 'classnames';
+import React, {useMemo} from 'react';
+
+import moduleStyles from './styles/validation-status-icon.module.scss';
+
+interface ValidationIconProps {
+  status: 'passed' | 'failed' | 'pending' | 'caution' | 'error';
+  className?: string;
+}
+
+// Component that extracts out the logic for rendering a validation status icon.
+// For any but the 'pending' icon, we use a duotone icon with a white x/check/minus/exclamation mark
+// and a colored circle background. For the 'pending' icon, we use a regular (unfilled) circle icon.
+const ValidationStatusIcon: React.FunctionComponent<ValidationIconProps> = ({
```

Comment:
nit: That's probably for the future, but still wanted to ask - Is there a plan to use DSCO `FontAwesomeIconV6` component some day maybe?
Though right now it might be challenging to render duotone icons with `FontAwesomeIconV6` - I'll take a look into it and make an update soon

Reply:
Yeah because I needed a bunch of extra configuration I used a generic icon.